src/sidebar/components/test/thread-list-omega-test.js

```diff
@@ -0,0 +1,280 @@
+      createComponent();
+
+      assert.notCalled(fakeScrollTop);
+    });
+
+    it('should do nothing if the annotation thread to scroll to is not in DOM', () => {
+      createComponent();
+
+      act(() => {
+        fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED](
+          {},
+          { $tag: 'nonexistent' }
+        );
+      });
+
+      assert.notCalled(fakeScrollTop);
+    });
+
+    it('should set the scroll container `scrollTop` to derived position of thread', () => {
+      createComponent();
+
+      act(() => {
+        fakeRootScope.eventCallbacks[events.BEFORE_ANNOTATION_CREATED](
+          {},
+          fakeTopThread.children[3].annotation
+        );
+      });
+
+      // The third thread in a collection of threads at default height (200)
+      // should be at 600px. This setting of `scrollTop` is the only externally-
+      // observable thing that happens here...
+      assert.calledWith(fakeScrollTop, 600);
+    });
+  });
+
+  describe('scroll and resize event handling', () => {
+    const debouncedFn = sinon.stub();
+
+    beforeEach(() => {
+      fakeDebounce.returns(debouncedFn);
+    });
+
+    it('updates scroll position and window height for recalculation on container scroll', () => {
+      createComponent();
+      document
+        .querySelector('.js-thread-list-scroll-root')
+        .dispatchEvent(new Event('scroll'));
+
+      assert.calledOnce(debouncedFn);
+    });
+
+    it('updates scroll position and window height for recalculation on window resize', () => {
+      createComponent();
+      window.dispatchEvent(new Event('resize'));
+
+      // TODO
+      assert.calledTwice(debouncedFn);
+    });
+  });
+
+  context('when top-level threads change', () => {
+    it('recalculates thread heights', () => {
+      const wrapper = createComponent();
+      // Initial render and effect hooks will trigger calculation twice
+      fakeDomUtil.getElementHeightWithMargins.resetHistory();
+      // Let's see it respond to the top-level threads changing
+      wrapper.setProps({ thread: fakeTopThread });
+      // It should check the element height for each top-level thread (assuming
+      // they are all onscreen, which these test threads "are")
+      assert.equal(
+        fakeDomUtil.getElementHeightWithMargins.callCount,
+        fakeTopThread.children.length
+      );
+    });
+  });
+
+  it('renders dimensional elements above and below visible threads', () => {
+    const wrapper = createComponent();
+    const upperDiv = wrapper.find('div').first();
+    const lowerDiv = wrapper.find('div').last();
+    assert.equal(upperDiv.getDOMNode().style.height, '400px');
+    assert.equal(lowerDiv.getDOMNode().style.height, '600px');
+  });
+
+  it('renders a `ThreadCard` for each visible thread', () => {
+    const wrapper = createComponent();
+    const cards = wrapper.find('ThreadCard');
+    assert.equal(cards.length, fakeTopThread.children.length);
+  });
+
+  // it(
```

Comment:
This `a11y` check will fail; it reports duplicate element IDs. I double-checked that this component doesn't render duplicate IDs...it doesn't...any advice?

Reply:
What _are_ the duplicate element IDs? Can you determine which component generated them?

Reply:
For posterity: this was a result of not removing the `fakeScrollContainer` in the `afterEach`—it was getting added multiple times.